apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts

```diff
@@ -60,20 +63,24 @@ export class VehicleService {
 
   async create(vehicle: VehicleModel): Promise<boolean> {
     try {
-      // Check if Vehicle is already in database
+      // check if vehicle is already in database
       const findVehicle = await this.findByVehicleId(vehicle.vehicleId)
+      // if vehicle is found in db, check if there is new owner of vehicle
       if (findVehicle) {
-        // Remove old request if new owner
         if (vehicle.ownerNationalId === findVehicle.ownerNationalId) {
           return true
         } else {
-          await findVehicle.destroy()
+          this.logger.info('vehicle has new owner, update national-id')
+          findVehicle.ownerNationalId = vehicle.ownerNationalId
+          await findVehicle.save()
+          return true
         }
+      } else {
+        await vehicle.save()
+        return true
       }
-      // Save vehicle to database
-      await vehicle.save()
-      return true
     } catch (err) {
+      this.logger.error('error creating vehicle', err)
```

Comment:
Mér var bent á að hafa loggerinn í template stíl um daginn. Sel það ekki dýrara en ég keypti það 

logger.error(
        `User ${user.nationalId} does not own the requested vehicle`,
        { permno: input.permno, user },
      )

Reply:
Ok en það er ekkert extra concat í gangi hér.

Reply:
Skil ekki alveg :)